message.go

```diff
@@ -688,7 +689,8 @@ func (d *Decoder) Decode() (*Message, error) {
 	}
 	maxSeg := SegmentID(binary.LittleEndian.Uint32(d.wordbuf[:]))
 	if maxSeg > maxStreamSegments {
-		return nil, errors.New("decode: too many segments to decode")
+		return nil, fmt.Errorf("decode: too many segments (%d) to decode (max=%d)",
```

Comment:
Per #364, can we void fmt here and use strconv instead?

Reply:
Sure.  Should I define a separate error type?

Reply:
You don't need to; you could also just concatenate strings and keep using errors.New. Your call.

Reply:
(You probably want to use internal/str.Utod instead of strconv directly)